tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -90,6 +95,22 @@ def frontend_metadata(self):
 
   def index_impl(self):
     """Return {runName: {tagName: {displayName: ..., description: ...}}}."""
+    if self._data_provider:
+      mapping = self._data_provider.list_scalars(
+          experiment_id=None,  # experiment support not yet implemented
+          owner_plugin=metadata.PLUGIN_NAME,
+      )
+      result = {run: {} for run in mapping}
+      for (run, tag_to_content) in six.iteritems(mapping):
+        for (tag, metadatum) in six.iteritems(tag_to_content):
+          sm = metadatum.summary_metadata
+          descr = plugin_util.markdown_to_safe_html(sm.summary_description)
+          result[run][tag] = {
+              'displayName': sm.display_name,
```

Comment:
For new code, it seems better to just use the tag going forward versus maintaining support for the display_name, since it's not in the new summary APIs (because it was not really the right solution to the problem it was introduced to solve).

Reply:
It seems fine to me to drop it.  It's in `tf.summary.write` only because we're re-using SummaryMetadata out of convenience.  None of the wrapper APIs allow passing a display_name and just because `write()` accepts one semi-indirectly doesn't mean it actually has to be honored.

I left in `description` since it seemed overall more of a correctly motivated feature (though perhaps still of limited use) and it somewhat softens the blow of removing display_name since then you can always just put anything meant to be human-oriented into the description.

Reply:
Out of curiosity, is there a reference describing why `display_name` is
not considered the right solution to the problem that it was introduced
to solve? Discussions on #59 and the linked TensorFlow issue don’t seem
to raise such a concern.

Reply:
If we do drop `display_name`, then I’d like to drop it across all of
TensorBoard for consistency, and preferably by the 2.0 release.
Otherwise, it’ll be a functional change from `--generic_data=false` to
`--generic_data=true`, and I’d like to minimize those.

(I’ll send a PR out for this.)

Reply:
Thanks for pointing out that the deprecation of `display_name` was never really documented anywhere in one place (but rather spread across bits of #1534, https://github.com/tensorflow/tensorflow/commit/92937fb439e102f31c26bd9234e03552499f6ba4, and internal documentation).

I've addressed that by wrapping up #59 with a long-overdue explanation: https://github.com/tensorflow/tensorboard/issues/59#issuecomment-521100225